src/lib.rs

```diff
@@ -448,7 +455,14 @@ impl Config {
                     if !self.defined("CMAKE_SYSTEM_NAME") {
                         self.define("CMAKE_SYSTEM_NAME", "Generic");
                     }
-                } else if target != host && !self.defined("CMAKE_SYSTEM_NAME") {
+                } else if target != host
+                    && !self.defined("CMAKE_SYSTEM_NAME")
+                    && !(msvc
+                        && self
+                            .generator
+                            .as_deref()
+                            .map_or(true, |g| g.to_string_lossy().starts_with("Visual Studio")))
```

Comment:
This is definitely weirdly complicated for an inline condition, and depends on the default generator choice for MSVC being Visual Studio (implemented below, unchanged). Suggestions welcome on how to reorganize this.